psi4/src/psi4/libdiis/diismanager.cc

```diff
@@ -213,53 +213,57 @@ bool DIISManager::add_entry(int numQuantities, ...) {
         if (i == _numErrorVectorComponents) arrayPtr = paramVector.data();
         switch (type) {
             case DIISEntry::InputType::Pointer:
+            {
                 array = va_arg(args, double *);
-                for (int j = 0; j < _componentSizes[i]; ++j) *arrayPtr++ = array[j];
+                auto size = _componentSizes[i];
+                std::copy(array, array + size, arrayPtr);
+                arrayPtr += size;
                 break;
+            }
             case DIISEntry::InputType::DPDBuf4:
+            {
                 buf4 = va_arg(args, dpdbuf4 *);
                 for (int h = 0; h < buf4->params->nirreps; ++h) {
                     global_dpd_->buf4_mat_irrep_init(buf4, h);
                     global_dpd_->buf4_mat_irrep_rd(buf4, h);
-                    for (int row = 0; row < buf4->params->rowtot[h]; ++row) {
-                        for (int col = 0; col < buf4->params->coltot[h]; ++col) {
-                            *arrayPtr++ = buf4->matrix[h][row][col];
-                        }
-                    }
+                    auto size = buf4->params->rowtot[h] * buf4->params->coltot[h ^ buf4->file.my_irrep];
```

Comment:
Is the use of `^ buf4->file.my_irrep` correct? I'm not seeing it in the original code.

Reply:
I am 99% sure that when @andysim created the code, he wasn't thinking about non totally-symmetric tensors. I grabbed these from the `axpy` functions of the original classes.

Reply:
I think you're right about that.  I wrote it for cumulants and Fock matrices, and likely wasn't forward-thinking enough to think about things like excited states.  The generalization to accommodate those methods is a good addition.